lib/web_ui/lib/src/engine/canvas_pool.dart

```diff
@@ -33,7 +33,8 @@ class _CanvasPool extends _SaveStackTracking {
 
   html.HtmlElement _rootElement;
   int _saveContextCount = 0;
-  int _activeCanvasCount = 0;
+  // Number of elements that have been added to flt-canvas.
+  int _activeElementCount = 0;
```

Comment:
Why not always have a non-null non-empty `_pool`? Then `_canvas` becomes `_pool.last`, and `_activeElementCount` becomes `_pool.length`, and we can remove all the null checks.

Reply:
pool only tracks canvas(s) if you have more than 1 canvas. activeElementCount tracks both canvas, p and img tags. So if you draw image and then canvas, activeElementCount is 2 at the end.